### ISSUE TITLE
About Classification Tasks

### ISSUE THREAD

**Issue by huifenghc** [bit-ml/DeCLIP#3]
hello,

First of all, great job on this impressive work! I really appreciate the effort and creativity that went into this project.
﻿
I wanted to ask if this model can be applied to classification tasks? The reason I'm asking is that I'm currently training a classification model, and I noticed a large performance gap between the training and validation sets. I'm curious if this work could potentially help address such issues.
﻿
Any insights or suggestions would be greatly appreciated!
﻿
Thanks again for sharing your work with the community!

The following is the log information.

Epoch mean train ACC: 84.27
Epoch mean train AP: 0.9842
Validation
Length of dataset: 226129
(Val @ epoch 0) ACC: 0.3363702595233917; AP: 0.9884438268397492
saving best model at the end of epoch 0
0.9884438268397492 0
Validation accuracy increased (-inf --> 0.336370).  Saving model ...

**Comment by MrNiceGuy090** [comment]
Hello! 

We did not experiment with image-level classification based on our model, but some results in this sense are listed in Wang et. al. [_OpenSDI: Spotting Diffusion-Generated Images in the Open World_](https://arxiv.org/abs/2503.19653). Ojha et. al. [_Towards Universal Fake Image Detectors that Generalize Across Generative Models_](https://arxiv.org/abs/2302.10174), one of the papers from which we started the idea of this work, is specifically designed for image-level detection, so I think this may be a better starting point. In addition, it uses the same backbone (CLIP) as our method. Although, the drop in performance on the test set may be due to the rather small size of the dolos benchmark (at least for image-level classification), thus training on a larger sample pool may help with the performance on the test set.

**Comment by huifenghc** [comment]
Well, thanks for your help.
